src/server.py

```diff
@@ -1,45 +1,151 @@
-            conn.sendall(full_string)
-            conn.sendall(response_ok())
+            try:
+                buffer_length = 8
+                message_complete = False
+                full_string = b""
+                while not message_complete:
+                    part = conn.recv(buffer_length)
+                    full_string = full_string + part
+                    if len(part) < buffer_length:
+                        full_string = full_string.decode('utf-8')
+                        message_complete = True
+                # print('full_string.decode', full_string)
+                try:
+                    uri = parse_request(full_string)
+                    body_tuple = resolve_uri(uri)
+                    if body_tuple:
+                        response = response_ok(body_tuple)
+                    else:
+                        response = response_error('404 Not Found')
+                    response = response_ok(body)
+                except SyntaxError:
+                    response = response_error('404 Page Not Found')
+                except NameError:
+                    response = response_error('405 Method Not Allowed')
+                except TypeError:
+                    response = response_error('505 HTTP Version Not Supported')
+                except ValueError:
+                    raise HTTPException(
+                        HTTP_BAD_REQUEST,
+                        "No 'Host :' in header")
+            except SystemError:
+                response = response_error()
+            # full_string += b"\r\n\r\n"
+            # conn.sendall(full_string)
+            conn.sendall(response.encode('utf-8'))
             conn.close()
     except KeyboardInterrupt:
         server.close()
 
 
-def response_ok():
-    """Return HTTP 200 OK message for when the connection is ok."""
-    return (b"HTTP/1.1 200 OK\r\n\r\n"
-            b"Success")
+def response_ok(body_tuple):
+    """Return 200 ok."""
+    first = 'HTTP/1.1 200 OK'
+    second_line = 'Content-Type:' + body_tuple[1] + '; charset=utf-8'
+    date = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header_break = ''
+    body = body_tuple
+    fourth_line = 'Content-Length: {}'.format(len(body))
+    string_list = [first, second_line, date, fourth_line, header_break, body]
+    string_list = '\r\n'.join(string_list) + '\r\n'
+    string_list = string_list.encode('utf-8') + body
+    return string_list
+
+
+def response_error(error='500 Internal Server Error'):
+    """Return 500 internal server error."""
+    first = 'HTTP/1.1 {}'.format(error)
+    second_line = 'Content-Type: text/plain; charset=utf-8'
+    date = email.utils.formatdate(usegmt=True)
+    header_break = ''
+    body = 'The system is down'
+    bytes_ = body.encode('utf-8')
+    fourth_line = 'Content-Length: {}'.format(len(bytes_))
+    string_list = [first, second_line, date, fourth_line, header_break, body]
+    string_list = '\r\n'.join(string_list)
+    return string_list
```

Comment:
The return value of `response_error` is now unicode, right?  But this means that you have to treat responses that are generated by `response_ok` above differently from responses generated here.  that's going to be complicated.  Can you simplify the problem by making both functions return the same data type?